src/validation.cpp

```diff
@@ -5291,6 +5291,262 @@ bool ExpectedAssumeutxo(const int height, uint256& expected_out, const CChainPar
     return false;
 }
 
+bool ChainstateManager::ActivateSnapshot(
+        CAutoFile& coins_file,
+        const SnapshotMetadata& metadata,
+        bool in_memory)
+{
```

Comment:
I'm not opposed, but I'm not sure I see a lot of benefit either. E.g. I don't think splitting this stuff up allows us to do a more comprehensive job of testing. I'm happy to apply a diff though.

Reply:
I tend to agree that (in the current incarnation 5e08b73c023bd7f9e79d1b67c25a2ed2b4248fe5 `PopulateAndValidateSnapshot` does quite a lot. Although I can follow along, it might be more clear if `ActivateSnapshot` called more specific helper functions in the right order, e.g. `WaitForHeaders`.